containers/entitlement-pdp/handlers/getEntitlements.go

```diff
@@ -66,59 +76,60 @@ type PDPEngine interface {
 // @Failure      404 {string} http.StatusNotFound
 // @Failure      500 {string} http.StatusServerError
 // @Router       /entitlements [post]
-func GetEntitlementsHandler(pdp PDPEngine, logger *zap.SugaredLogger) http.Handler {
-	entitlementsHandler := func(w http.ResponseWriter, req *http.Request) {
-		spanCtx := req.Context()
-		handlerCtx, span := tracer.Start(spanCtx, "GetEntitlementsHandler")
-		defer span.End()
-
-		if req.Method != http.MethodPost {
-			logger.Error("Rejected request, invalid HTTP verb")
-			http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
-			return
-		}
+func (e Entitlements) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	spanCtx := req.Context()
+	handlerCtx, span := tracer.Start(spanCtx, "GetEntitlementsHandler")
+	defer span.End()
 
-		logger.Debug("GetEntitlementsHandler - reading request body")
+	if req.Method != http.MethodPost {
+		e.Logger.Error("Rejected request, invalid HTTP verb")
+		http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+		return
+	}
 
-		//Read + consume body
-		bodBytes, err := io.ReadAll(req.Body)
-		defer req.Body.Close()
-		if err != nil {
-			logger.Errorf("Couldn't read client request! Error was %s", err)
-			w.WriteHeader(http.StatusBadRequest)
-			return
-		}
+	e.Logger.Debug("GetEntitlementsHandler - reading request body")
 
-		payload, err := getRequestPayload(bodBytes, handlerCtx, logger)
+	// Read + consume body
+	bodBytes, err := io.ReadAll(req.Body)
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
 		if err != nil {
-			logger.Errorf("Couldn't deserialize client request! Error was %s", err)
-			w.WriteHeader(http.StatusBadRequest)
-			return
+			log.Println(err)
```

Comment:
Why are you not using the zap logger here?
```suggestion
			e.Logger.Error(err)
```

Reply:
bad habit, this needs to be cleaned up